src/Native/Unix/configure.cmake

```diff
@@ -572,14 +564,12 @@ check_symbol_exists(
     HAVE_TCP_FSM_H
 )
 
-check_cxx_source_compiles(
-    "
-    #include <sys/types.h>
-    #include <net/route.h>
-    int main() { rt_msghdr* hdr; return 0; }
-    "
-    HAVE_RT_MSGHDR
-)
+set(CMAKE_EXTRA_INCLUDE_FILES sys/types.h net/route.h)
+check_type_size(
+    "struct rt_msghdr"
+     HAVE_RT_MSGHDR
+     BUILTIN_TYPES_ONLY)
+set(CMAKE_EXTRA_INCLUDE_FILES) # reset CMAKE_EXTRA_INCLUDE_FILES
```

Comment:
For every one of these check_* that's being modified, please review the CI output for both a Linux and macOS leg to confirm that nothing changes before/after with regards to what cmake configuration determines.

Reply:
As part of this work, I have done that from the get go by taking diffs with other PRs.